src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -1314,6 +1316,78 @@ private void AddPythonEnginesOptions()
                 options.Add(item.Name);
             }
             PythonEnginesList = options;
+            PopulatePythonEnginePackageMap();
+        }
+
+        private void PopulatePythonEnginePackageMap()
+        {
+            var packageBinPaths = new List<string>();
+            foreach (var packagePath in preferenceSettings.CustomPackageFolders)
+            {
+                var ppath = packagePath == DynamoModel.BuiltInPackagesToken ? PathManager.BuiltinPackagesDirectory : packagePath;
+                packageBinPaths = packageBinPaths.Concat(Directory.GetFiles(ppath, "*.dll", SearchOption.AllDirectories)).ToList();
+            }
+
+            var dotNetRuntimePaths = Directory.GetFiles(RuntimeEnvironment.GetRuntimeDirectory(), "*.dll");
+            var dynCorePaths = new FileInfo(Assembly.GetExecutingAssembly().Location).Directory.GetFiles("*.dll", SearchOption.AllDirectories).Select(x => x.FullName);
+            var standardPaths = dotNetRuntimePaths.Concat(dynCorePaths);
+
+            MetadataLoadContext mlc = null;
+            var resolver = new PathAssemblyResolver(standardPaths);
+            mlc = new MetadataLoadContext(resolver);
+            var all = AppDomain.CurrentDomain.GetAssemblies().ToList();
```

Comment:
I worry about this because you'll only find assemblies loaded into the default ALC, I don't think this will let you see all the assemblies loaded into the current process. You may need to dig around in the dotnet source to confirm or deny that.

`AppDomain` is now just a facade on top of assembly load contexts.